### DISCUSSION TITLE
Using Crashlytics for multiple firebase apps

### DISCUSSION THREAD

**Discussion by eugene-kordin** [invertase/react-native-firebase#5592] [Q&A]
Does anybody can confirm this library supports multiple Firebase App? According to the guides from Google, the google-services.json file should be deleted from the project to setup Firebase Apps in the runtime. But looks like during the import like this:  
`import crashlytics from '@react-native-firebase/crashlytics'`

Crashlytics wants to initialize a default App from the json file that does not exist. I am not able to use this import at the top of the file.

If I am not using this import an access to Crashlytics from the App instance also fails:

```
app.crashlytics.setCrashlyticsCollectionEnabled(true);
An unhandled error was caught from submitForm() TypeError: app.crashlytics.setCrashlyticsCollectionEnabled is not a function
    at authApp (VM5 index.bundle:82)
```

Is this feature supported or not? Maybe some additional configuration is required?

**Comment by mikehardy** [comment]
Crashlytics definitely does not support multiple apps

https://rnfirebase.io/app/usage#secondary-apps

**Reply by eugene-kordin** [reply]
Thanks, but does it support the manual setup of the single App instance from the parameters (not from default constructor that will use google-services.json file on disk? Can I have the following in my code and use Crashlytics? Or it is not supported? 

```
export const config = {
  apiKey: 'A000000000000000',
  authDomain: 'test1au.firebaseapp.com',
  databaseURL: 'https://test1au.firebaseio.com',
  projectId: 'test1au',
  storageBucket: 'tsome.appspot.com',
  messagingSenderId: '9900000099',
  appId: '1:XXXXXXXXX9:android:26000000006',
}
firebase.initializeApp(config)
```

Because in such case I see an error I described if I import Crashlytics module.

**Reply by mikehardy** [reply]
I don't believe that is supported - the issue with crashlytics is that it needs to be initialized long long before the JS thread exists and doing any sort of configuration. The only way to do it without the google-services.json file is to do the work that the google-service-plugin + google-services.json file was doing, that is: inject the information directly in your app via xml resources, as described here: https://developers.google.com/android/guides/google-services-plugin

I can't think of why this would be preferred, but perhaps it is?

Configuration of crashlytics in the javascript layer is too late though
